.github/workflows/test_build_docker_image.yaml

```diff
@@ -0,0 +1,62 @@
+name: "Test build Docker image"
```

Comment:
Had a similar idea as well, but decided against it to avoid confusion. We already have a job called `build_and_push_docker_image` that's part of `release.yaml`. This job was part of `tests.yaml`, so I prefixed it with "Test" to see which is which. AFAICT, it also doesn't push anything because of `push: false`, so looks like the intention here is to just test whether a docker container can be built.

In the interest of time, I'm going to merge this as is since it's already approved, but this can always be resolved in followup PRs.

Reply:
But it is not testing the docker image just building it the fact that we have a step/job named build and push is inconsequential to having a workflow named build or build and push.

Reply:
Yeah, I think a better name might be "Build Docker image", but we already have a file called `build_docker_image.yaml` with the name "Publish Dev Docker Images".

I think my preferred solution would be making all these consistent, so that the name matches the filename and there are no clashes between files. Something like this (untested):
- "Publish dev Docker image" -> publish_dev_docker_image.yaml (renamed from "Publish Dev Docker Images")
- "Build Docker image" -> build_docker_image.yaml (the workflow discussed in this thread)

@peytondmurray Hey Peyton, I'm currently on PTO, would you be comfortable taking over this in a separate PR? The task would be coming up with a consistent scheme for names and filenames of docker workflows.